encoders/nlp/UniversalSentenceEncoder/__init__.py

```diff
@@ -30,11 +34,28 @@ def __init__(
 
     def post_init(self):
         self.to_device()
+        if UNIVERSAL_SENTENCE_ENCODER_CMLM == self.model_url:
+            self.model = self._load_model_cmlm
+
+        else:
+            self.model = self._load_model_universal_sentence
+
+    @as_ndarray
+    def _load_model_cmlm(self, data: 'np.ndarray') -> 'np.ndarray':
```

Comment:
can't this be unified? Just having an encodrr and an optional preprocessor and have the user input them?

Reply:
do you talk about  avoiding  these multiple functions and reuse the .load function? If it is this, I can do it and if you want but I am not sure if it would open for an extension (close-open principle), I am only encapsulating your different use cases when you are initializing a model. Anyway, it doesn't exit more use cases (we need preprocess and we don't need it), it seems easier solution.

Reply:
but the point is to have this as a wrapper for this KerasLayer and preprocessing (if possible) and then have the user choose which encoder from TF Hub can be used and which preprocessor (if any). 

I am not sure about the details but I undersatnd this pattern should work for different encoders of the same family.

Reply:
I will change it... It is not related with a pattern, it is more a design principle, but the point is that you have clear your requirement, you want to specify your preprocessor outside an "optional" parameter, then the natural way is to do it as you want.